routing.go

```diff
@@ -336,8 +330,8 @@ func (dht *IpfsDHT) getValues(ctx context.Context, key string, stopQuery chan st
 
 	go func() {
 		defer close(valCh)
-		defer close(queriesCh)
-		queries, err := dht.runDisjointQueries(ctx, dht.d, key,
+		defer close(lookupResCh)
```

Comment:
@petar these are examples of the channels that we'd like to close once the query completes. There are other coding approaches we can take to handle this closure issue, but for now waiting for the query goroutines to shutdown seems like the safest option.